js/challenges/001_EvenNumbers/index.mjs

```diff
@@ -0,0 +1,11 @@
+/* 
+ * Returns a new array with only the odd numbers from the original, if any.
+ * @param {array} arr Original Array
+ * @returns {array} New odd-only array
+ */
+export default (arr) => {
+    return arr.filter((num) => {
+        if (isNaN(num)) throw new TypeError('Original array must contain only numbers.');
+        if (num % 2 !== 0) return true;
+    });
```

Comment:
**Interview Follow-up Question**:

Can you compare the use of the filter function with traditional iterative approaches like `for` and `while` loops? What are the advantages or benefits of using filter over these classic iteration methods?

Reply:
Filter really is doing the work of creating a new array, iterating through the old one, evaluating the conditional, and adding the right values to the new array.  When you understand what filter does, and what needs to be done, I think it is more concise, faster to write, and readable.  I just went and did some research though, and the for loop would be faster, because it is happening under the hood anyways, just with extra steps to wrap that functionality under the name of filter.